lib/ansible/module_utils/powershell/Ansible.ModuleUtils.Backup.psm1

```diff
@@ -0,0 +1,27 @@
+# Copyright (c): 2018, Dag Wieers (@dagwieers) <dag@wieers.com>
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+Function Backup-File($path, $obj=@{}) {
+<#
+    .SYNOPSIS
+    Helper function to make a backup of a file.
+    .EXAMPLE
+    Backup-File -path $path -obj $result
+#>
+    [CmdletBinding(SupportsShouldProcess=$true)]
+
+    $backup_path = $null
+    if (Test-Path -LiteralPath $path) {
+        $backup_path = "$path.$pid." + [DateTime]::Now.ToString("yyyyMMdd-HHmmss") + ".bak";
+        Try {
+            Copy-Item -LiteralPath $path -Destination $backup_path
```

Comment:
Sorry, it looks like I was wrong about `-WhatIf` propagating to downstream cmdlets. We can either re-add a separate parameter like you had before or change this line to `Copy-Item -LiteralPath $path -Destination $backup_path -WhatIf:$WhatIfPreference`.

I'm on the side of keeping things more PowerShell like which is to reuse the `-WhatIf` idioms but happy to be convinced otherwise.

Reply:
Fixed.